src/ru/nsu/ccfit/zuev/osu/game/GameScene.java

```diff
@@ -2100,6 +2104,19 @@ private String registerHit(final int objectId, final int score, final boolean en
         return scoreName;
     }
 
+    private void onObjectHit(final int score){
+        // mod or config's logic on note hit
+        if(GameHelper.isSuddenDeath() && score == 0){
+            stat.changeHp(-1.0f);
+        }
+        if(GameHelper.isPerfect() && (score == 100 || score == 50 || score == 0)){
+            gameover();
+            restartGame();
+        }
+        if(Config.isRemoveNoteLockWhenMiss()){
+            Config.setRemoveNoteLock(score == 0);
+        }
```

Comment:
Having this set in `Config` reads a bit weird. Maybe have this as a private field in `GameScene` instead?

Reply:
> Having this set in `Config` reads a bit weird. Maybe have this as a private field in `GameScene` instead?

maybe I will add a RemoveNoteLock option, so it's in `Config`